### ISSUE TITLE
possibility to run BigDL with dynamic resource allocation

### ISSUE THREAD

**Issue by chen116** [intel-analytics/BigDL#2796]
Hi,

As of now, BigDL is not able to run with dynamic resource allocation ( #879 ). is there any way to make it possible? I wish to have such ability for BigDL so I can test out my new mechanism for dynamic executor allocation for spark streaming application.

Thanks

**Comment by wzhongyuan** [comment]
Yes, current BigDL design does not support dynamic resource allocation.  could you please light more details on your scenario ?

**Comment by chen116** [comment]
HI, Thanks for the response.

So the reasons I wish to run dynamic resource allocation with BigDL is because:

1. I want to make spark streaming application meeting certain timing performance dynamically and with efficient resource usage so I modified the current spark dynamic resource allocation algorithm to dynamically change the number of executors based on the previous batches' timing performance. 

2. To test out my custom dynamic resource allocation, I searched and found this spark streaming benchmark that uses BigDL (https://github.com/vmware/iot-analytics-benchmark/tree/master/DL) to execute image classification. 

3. By running the benchmark with different number of executor statically, I can clearly see the timing performance is better when more executors are used. So my next step is to plug in my dynamic resource allocation module to see if it works in real-time. But then when I enable dynamic resource allocation, I got the error message about not supporting dynamic resource allocation.

Thanks again for your time

also,
your sentence , "**Yes, current BigDL design does not support dynamic resource allocation**." is confusing to me. Does the "yes" means there is a way use dynamic resource allocation? if so, it is contradicted by following statement "current BigDL design does not support dynamic resource allocation." can you please clarify? thank you

**Comment by wzhongyuan** [comment]
Sorry, I mean "No,  currently BigDL design does not support dynamic resource allocation".
you may take BigDL as a normal Java library and not use those RDD specific apis in your case. 
Probably you can show a minimal piece of code that can reproduce the issue.
